cppapi/server/fwdattrdesc.cpp

```diff
@@ -180,7 +181,13 @@ bool FwdAttr::validate_fwd_att(vector<AttrProperty> &prop_list,const string &dev
 	}
 	catch (...) {}
 
-	if (root_att_db_defined == true)
+	//check if full_root_att is already set
+	if (full_root_att.size()!=0 && full_root_att.compare(RootAttNotDef)!=0)
```

Comment:
Ah, I did not notice the last `else`. You are right then: if `is_full_root_att_set == true` `full_root_att` is set to `RootAttNotDef`. In this case just wrap inner `if/else` with outer:

```c++
if(!is_full_root_att_set){
    if(root_att_db_defined)
        full_root_att = root_att_db;
    else
        full_root_att = RootAttNotDef;
}
```

The later I find better than empty statement.